File: android/sources/src/com/applifier/impact/android/ApplifierImpact.java
package com.applifier.impact.android;

import java.util.ArrayList;
import java.util.Map;

import com.unity3d.ads.android.IUnityAdsListener;
import com.unity3d.ads.android.UnityAds;

import android.app.Activity;
import android.os.Build;

public class ApplifierImpact implements IUnityAdsListener {
	
	// Reward item HashMap keys
	public static final String APPLIFIER_IMPACT_REWARDITEM_PICTURE_KEY = "picture";
	public static final String APPLIFIER_IMPACT_REWARDITEM_NAME_KEY = "name";
	
	// Impact developer options keys
	public static final String APPLIFIER_IMPACT_OPTION_NOOFFERSCREEN_KEY = "noOfferScreen";
	public static final String APPLIFIER_IMPACT_OPTION_OPENANIMATED_KEY = "openAnimated";
	public static final String APPLIFIER_IMPACT_OPTION_GAMERSID_KEY = "sid";
	public static final String APPLIFIER_IMPACT_OPTION_MUTE_VIDEO_SOUNDS = "muteVideoSounds";
	public static final String APPLIFIER_IMPACT_OPTION_VIDEO_USES_DEVICE_ORIENTATION = "useDeviceOrientationForVideo";

	// Impact components
	public static ApplifierImpact instance = null;
	
	// Listeners
	private IApplifierImpactListener _impactListener = null;
	
	
	public ApplifierImpact (Activity activity, String gameId) {
		init(activity, gameId, null);
	}
	
	public ApplifierImpact (Activity activity, String gameId, IApplifierImpactListener listener) {
		init(activity, gameId, listener);
	}

	private void init (final Activity activity, String gameId, IApplifierImpactListener listener) {
		instance = this;
		setImpactListener(listener);
		UnityAds.init(activity, gameId, this);
	}

	
	/* PUBLIC STATIC METHODS */
	
	public static boolean isSupported () {
		if (Build.VERSION.SDK_INT < 9) {
			return false;
		}
		
		return true;
	}
	
	public static void setDebugMode (boolean debugModeEnabled) {
		UnityAds.setDebugMode(debugModeEnabled);
	}
	
	public static void setTestMode (boolean testModeEnabled) {
		UnityAds.setTestMode(testModeEnabled);
	}
	
	public static void setTestDeveloperId (String testDeveloperId) {
		UnityAds.setTestDeveloperId(testDeveloperId);
	}
	
	public static void setTestOptionsId (String testOptionsId) {
		UnityAds.setTestOptionsId(testOptionsId);
	}
	
	public static String getSDKVersion () {
		return UnityAds.getSDKVersion();
	}
	
	
	/* PUBLIC METHODS */
	
	public void setImpactListener (IApplifierImpactListener listener) {
		_impactListener = listener;
	}
	
	public void changeActivity (Activity activity) {
		UnityAds.changeActivity(activity);
	}
	
	public boolean hideImpact () {
		return UnityAds.hide();
	}
	
	public boolean setZone(String zoneId) {
		return UnityAds.setZone(zoneId);
	}
	
	public boolean setZone(String zoneId, String rewardItemKey) {
		return UnityAds.setZone(zoneId, rewardItemKey);
	}
	
	public boolean showImpact (Map<String, Object> options) {
		return UnityAds.show(options);
	}
	
	public boolean showImpact () {
		return UnityAds.show();
	}
	
	public boolean canShowCampaigns () {
		return UnityAds.canShowAds();
	}
	
	public boolean canShowImpact () {
		return UnityAds.canShow();
	}

	public void stopAll () {
	}
	
	
	/* PUBLIC MULTIPLE REWARD ITEM SUPPORT */
	
	public boolean hasMultipleRewardItems () {
		return UnityAds.hasMultipleRewardItems();
	}
	
	public ArrayList<String> getRewardItemKeys () {
		return UnityAds.getRewardItemKeys();
	}
	
	public String getDefaultRewardItemKey () {
		return UnityAds.getDefaultRewardItemKey();
	}
	
	public String getCurrentRewardItemKey () {
		return UnityAds.getCurrentRewardItemKey();
	}
	
	public boolean setRewardItemKey (String rewardItemKey) {
		return UnityAds.setRewardItemKey(rewardItemKey);
	}
	
	public void setDefaultRewardItemAsRewardItem () {
		UnityAds.setDefaultRewardItemAsRewardItem();
	}
	
	public Map<String, String> getRewardItemDetailsWithKey (String rewardItemKey) {
		return UnityAds.getRewardItemDetailsWithKey(rewardItemKey);
	}
	
	@Override
	public void onHide() {
		if (_impactListener != null)
			_impactListener.onImpactClose();
	}

	@Override
	public void onShow() {
		if (_impactListener != null)
			_impactListener.onImpactOpen();
	}

	@Override
	public void onVideoStarted() {
		if (_impactListener != null) {
			_impactListener.onVideoStarted();
		}
	}

	@Override
	public void onVideoCompleted(String rewardItemKey, boolean skipped) {
		if (_impactListener != null) {
			_impactListener.onVideoCompleted(rewardItemKey, skipped);
		}
	}

	@Override
	public void onFetchCompleted() {
		if (_impactListener != null) {
			_impactListener.onCampaignsAvailable();
		}
	}

	@Override
	public void onFetchFailed() {
		if (_impactListener != null) {
			_impactListener.onCampaignsFetchFailed();
		}
	}
}

File: ios/UnityAds/ApplifierImpact.m
//
//  ApplifierImpact.m
//  Copyright (c) 2012 Unity Technologies. All rights reserved.
//

#import "ApplifierImpact.h"
#import <UnityAds.h>

NSString * const kApplifierImpactRewardItemPictureKey = @"picture";
NSString * const kApplifierImpactRewardItemNameKey = @"name";

NSString * const kApplifierImpactOptionNoOfferscreenKey = @"noOfferScreen";
NSString * const kApplifierImpactOptionOpenAnimatedKey = @"openAnimated";
NSString * const kApplifierImpactOptionGamerSIDKey = @"sid";
NSString * const kApplifierImpactOptionMuteVideoSounds = @"muteVideoSounds";
NSString * const kApplifierImpactOptionVideoUsesDeviceOrientation = @"useDeviceOrientationForVideo";

@interface ApplifierImpact () <UnityAdsDelegate>
  @property (nonatomic, assign) Boolean debug;
@end

@implementation ApplifierImpact

#pragma mark - Static accessors

+ (BOOL)isSupported {
  return [UnityAds isSupported];
}

- (void)setTestDeveloperId:(NSString *)developerId {
  [[UnityAds sharedInstance] setTestDeveloperId:developerId];
}

- (void)setTestOptionsId:(NSString *)optionsId {
  [[UnityAds sharedInstance] setTestOptionsId:optionsId];
}

+ (NSString *)getSDKVersion {
  return [UnityAds getSDKVersion];
}

- (void)setDebugMode:(BOOL)debugMode {
  [[UnityAds sharedInstance] setDebugMode:debugMode];
}

- (BOOL)isDebugMode {
  return [[UnityAds sharedInstance] isDebugMode];
}

static ApplifierImpact *sharedImpact = nil;

+ (ApplifierImpact *)sharedInstance {
	@synchronized(self) {
		if (sharedImpact == nil) {
      sharedImpact = [[ApplifierImpact alloc] init];
      sharedImpact.debug = NO;
      [[UnityAds sharedInstance] setDelegate:sharedImpact];
		}
	}
	
	return sharedImpact;
}

#pragma mark - Public

- (void)setTestMode:(BOOL)testModeEnabled {
  [[UnityAds sharedInstance] setTestMode:testModeEnabled];
}

- (BOOL)startWithGameId:(NSString *)gameId {
  return [[UnityAds sharedInstance] startWithGameId:gameId];
}

- (BOOL)startWithGameId:(NSString *)gameId andViewController:(UIViewController *)viewController {
  return [[UnityAds sharedInstance] startWithGameId:gameId andViewController:viewController];
}

- (BOOL)canShowCampaigns {
  return [[UnityAds sharedInstance] canShowAds];
}

- (BOOL)canShowImpact {
  return [[UnityAds sharedInstance] canShowAds];
}

- (BOOL)setZone:(NSString *)zoneId {
  return [[UnityAds sharedInstance] setZone:zoneId];
}

- (BOOL)setZone:(NSString *)zoneId withRewardItem:(NSString *)rewardItemKey {
  return [[UnityAds sharedInstance] setZone:zoneId withRewardItem:rewardItemKey];
}

- (BOOL)showImpact:(NSDictionary *)options {
  return [[UnityAds sharedInstance] show:options];
}

- (BOOL)showImpact {
  return [[UnityAds sharedInstance] show];
}

- (BOOL)hasMultipleRewardItems {
  return [[UnityAds sharedInstance] hasMultipleRewardItems];
}

- (NSArray *)getRewardItemKeys {
  return [[UnityAds sharedInstance] getRewardItemKeys];
}

- (NSString *)getDefaultRewardItemKey {
  return [[UnityAds sharedInstance] getDefaultRewardItemKey];
}

- (NSString *)getCurrentRewardItemKey {
  return [[UnityAds sharedInstance] getCurrentRewardItemKey];

}

- (BOOL)setRewardItemKey:(NSString *)rewardItemKey {
  return [[UnityAds sharedInstance] setRewardItemKey:rewardItemKey];
}

- (void)setDefaultRewardItemAsRewardItem {
  [[UnityAds sharedInstance] setDefaultRewardItemAsRewardItem];
}

- (NSDictionary *)getRewardItemDetailsWithKey:(NSString *)rewardItemKey {
  return [[UnityAds sharedInstance] getRewardItemDetailsWithKey:rewardItemKey];
}

- (BOOL)hideImpact {
  return [[UnityAds sharedInstance] hide];
}

- (void)setViewController:(UIViewController *)viewController showImmediatelyInNewController:(BOOL)applyImpact {
  [[UnityAds sharedInstance] setViewController:viewController];
}

- (void)stopAll{
}

#pragma mark - UnityAdsDelegate

- (void)unityAdsWillShow {
  if (self.delegate != nil && [self.delegate respondsToSelector:@selector(applifierImpactWillOpen:)]) {
    [self.delegate applifierImpactWillOpen:self];
  }
}

- (void)unityAdsDidShow {
  if (self.delegate != nil && [self.delegate respondsToSelector:@selector(applifierImpactDidOpen:)]) {
    [self.delegate applifierImpactDidOpen:self];
  }
}

- (void)unityAdsWillHide {
  if (self.delegate != nil && [self.delegate respondsToSelector:@selector(applifierImpactWillClose:)]) {
    [self.delegate applifierImpactWillClose:self];
  }
}

- (void)unityAdsDidHide {
  if (self.delegate != nil && [self.delegate respondsToSelector:@selector(applifierImpactDidClose:)]) {
    [self.delegate applifierImpactDidClose:self];
  }
}

- (void)unityAdsWillLeaveApplication {
  if (self.delegate != nil && [self.delegate respondsToSelector:@selector(applifierImpactWillLeaveApplication:)]) {
    [self.delegate applifierImpactWillLeaveApplication:self];
  }
}

- (void)unityAdsVideoStarted {
  if (self.delegate != nil && [self.delegate respondsToSelector:@selector(applifierImpactVideoStarted:)]) {
    [self.delegate applifierImpactVideoStarted:self];
  }
}

- (void)unityAdsFetchCompleted {
  if (self.delegate != nil && [self.delegate respondsToSelector:@selector(applifierImpactCampaignsAreAvailable:)]) {
    [self.delegate applifierImpactCampaignsAreAvailable:self];
  }
}

- (void)unityAdsFetchFailed {
  if (self.delegate != nil && [self.delegate respondsToSelector:@selector(applifierImpactCampaignsFetchFailed:)]) {
    [self.delegate applifierImpactCampaignsFetchFailed:self];
  }
}

- (void)unityAdsVideoCompleted:(NSString *)rewardItemKey skipped:(BOOL)skipped {
  if (self.delegate != nil) {
    [self.delegate applifierImpact:self completedVideoWithRewardItemKey:rewardItemKey videoWasSkipped:skipped];
  }
}

@end